internal/services/recommendation_poller.go

```diff
@@ -0,0 +1,141 @@
+		}
+		log.Errorf("unable to list recommendation for: %v", err)
+	}
+
+	// TODO: Is_valid_recommendation to be called on every container record v20.1 upgrade on wards
+	if kruize.Is_valid_recommendation(recommendation) {
+		containers := recommendation[0].Kubernetes_objects[0].Containers
+		for _, container := range containers {
+			for _, v := range container.Recommendations.Data {
+				marshalData, err := json.Marshal(v)
+				if err != nil {
+					log.Errorf("unable to list recommendation for: %v", err)
+				}
+
+				// Create RecommendationSet entry into the table.
+				recommendationSet := model.RecommendationSet{
+					WorkloadID:          kafkaMsg.Metadata.Workload_id,
+					ContainerName:       container.Container_name,
+					MonitoringStartTime: v.Duration_based.Short_term.Monitoring_start_time,
+					MonitoringEndTime:   v.Duration_based.Short_term.Monitoring_end_time,
+					Recommendations:     marshalData,
+				}
+				if err := recommendationSet.CreateRecommendationSet(); err != nil {
+					log.Errorf("unable to save a record into recommendation set: %v. Error: %v", recommendationSet, err)
+				} else {
+					log.Infof(" %s - Recommendation saved for experiment - %s and end_interval - %s", recommendationType, experiment_name, recommendationSet.MonitoringEndTime)
+				}
+
+				// Create entry into HistoricalRecommendationSet table.
+				historicalRecommendationSet := model.HistoricalRecommendationSet{
+					OrgId:               kafkaMsg.Metadata.Org_id,
+					WorkloadID:          kafkaMsg.Metadata.Workload_id,
+					ContainerName:       container.Container_name,
+					MonitoringStartTime: v.Duration_based.Short_term.Monitoring_start_time,
+					MonitoringEndTime:   v.Duration_based.Short_term.Monitoring_end_time,
+					Recommendations:     marshalData,
+				}
+				if err := historicalRecommendationSet.CreateHistoricalRecommendationSet(); err != nil {
+					recommendationJSON, _ := json.Marshal(recommendation)
+					log.Errorf("unable to get or add record to historical recommendation set table: %s. Error: %v", string(recommendationJSON), err)
+				}
+
+			}
+		}
+		poll_cycle_complete = true
+	} else {
+		poll_cycle_complete = true
+		invalidRecommendation.Inc()
+	}
+	return poll_cycle_complete
+}
+
+func PollForRecommendations(msg *kafka.Message, consumer_object *kafka.Consumer) {
+	log := logging.GetLogger()
+	validate := validator.New()
+	var kafkaMsg types.RecommendationKafkaMsg
+
+	if !json.Valid([]byte(msg.Value)) {
+		log.Errorf("received message on kafka topic is not vaild JSON: %s", msg.Value)
+		commitKafkaMsg(msg, consumer_object)
+		return
+	}
+	if err := json.Unmarshal(msg.Value, &kafkaMsg); err != nil {
+		log.Errorf("unable to decode kafka message: %s", msg.Value)
+		commitKafkaMsg(msg, consumer_object)
+		return
+	}
+	if err := validate.Struct(kafkaMsg); err != nil {
+		log.Errorf("invalid kafka message: %s", err)
+		commitKafkaMsg(msg, consumer_object)
+		return
```

Comment:
@patilsuraj767 Would these `return`s stop the consumer instance? I have replicated these from the processor.
Looking for a mechanism to stop code flow post `commitKafkaMsg` in this handler.